bindings/ocaml/SharedFunctors.ml

```diff
@@ -190,6 +190,43 @@ module Make_HKDF_generic (C: Buffer)
     Impl.extract (C.ctypes_buf prk) (C.ctypes_buf salt) (C.size_uint32 salt) (C.ctypes_buf ikm) (C.size_uint32 ikm)
 end
 
+module Make_ECDSA_generic (C: Buffer)
+    (Impl : sig
+       val min_msg_size : int
+       val sign : C.buf -> uint32 -> C.buf -> C.buf -> C.buf -> uint64
+       val verify : uint32 -> C.buf -> C.buf -> C.buf -> C.buf -> bool
+     end)
+= struct
+  type t = C.t
+  let get_result r =
+    if r = UInt64.zero then
+      true
+    else
+    if r = UInt64.max_int then
+      false
+    else
+      failwith "Unknown return value"
+  let prime_p256_order = Z.of_string "115792089210356248762697446949407573529996955224135760342422259061068512044369"
+  let sign priv msg k signature =
+    (* Hacl.Interface.P256.ECDSA.ecdsa_sign_p256_without_hash/sha2/sha384 *)
+    assert (C.size signature = 64);
```

Comment:
@Frosne can you check the defensive checks here?

Reply:
Done. I am fine with it :)